fabric-convention-tags-v1/src/client/java/net/fabricmc/fabric/impl/tag/convention/client/ConventionLogWarnings.java

```diff
@@ -0,0 +1,216 @@
+	}
+
+	private static void setupUntranslatedItemTagWarning() {
+		// Log missing item tag translations only in development environment and not running dedicated server.
+		ServerLifecycleEvents.SERVER_STARTED.register(server -> {
+			boolean isConfigSetToDev =
+					LOG_UNTRANSLATED_WARNING_MODE == LOG_WARNING_MODES.DEV_SHORT
+					|| LOG_UNTRANSLATED_WARNING_MODE == LOG_WARNING_MODES.DEV_VERBOSE;
+
+			if (FabricLoader.getInstance().isDevelopmentEnvironment() == isConfigSetToDev) {
+				Registry<Item> itemRegistry = server.getRegistryManager().get(RegistryKeys.ITEM);
+				List<TagKey<Item>> untranslatedItemTags = new ObjectArrayList<>();
+				itemRegistry.streamTags().forEach(itemTagKey -> {
+					// We do not translate vanilla's tags at this moment.
+					if (itemTagKey.id().getNamespace().equals("minecraft")) {
+						return;
+					}
+
+					String translationKey = TagUtil.getTagTranslationKey(itemTagKey);
+
+					if (!I18n.hasTranslation(translationKey)) {
+						untranslatedItemTags.add(itemTagKey);
+					}
+				});
+
+				if (!untranslatedItemTags.isEmpty()) {
+					StringBuilder stringBuilder = new StringBuilder();
+					stringBuilder.append("""
+							\n	Dev warning - Untranslated Item Tags detected. Please translate your item tags so other mods such as recipe viewers can properly display your tag's name.
+								You can disable this dev message in Fabric API's properties config file by setting log-untranslated-item-tag-warning to "SILENCED" or see individual tags with "DEV_VERBOSE".
+							""");
+
+					// Print out all untranslated tags when desired.
+					boolean isConfigSetToVerbose =
+							LOG_UNTRANSLATED_WARNING_MODE == LOG_WARNING_MODES.DEV_VERBOSE
+							|| LOG_UNTRANSLATED_WARNING_MODE == LOG_WARNING_MODES.PROD_VERBOSE;
+
+					if (isConfigSetToVerbose) {
+						stringBuilder.append("\nUntranslated item tags:");
+
+						for (TagKey<Item> tagKey : untranslatedItemTags) {
+							stringBuilder.append("\n     ").append(tagKey.id());
+						}
+					}
+
+					LOGGER.warn(stringBuilder.toString());
+				}
+			}
+		});
+	}
+
+	// Remove in 1.22
+	private static void setupLegacyTagWarning() {
+		// Log tags that are still using legacy 'c' namespace
+		ServerLifecycleEvents.SERVER_STARTED.register(server -> {
+			boolean isConfigSetToDev =
+					LOG_LEGACY_WARNING_MODE == LOG_WARNING_MODES.DEV_SHORT
+					|| LOG_LEGACY_WARNING_MODE == LOG_WARNING_MODES.DEV_VERBOSE;
+
+			if (FabricLoader.getInstance().isDevelopmentEnvironment() == isConfigSetToDev) {
+				List<TagKey<?>> legacyTags = new ObjectArrayList<>();
+				DynamicRegistryManager.Immutable dynamicRegistries = server.getRegistryManager();
+
+				// We only care about vanilla registries
+				dynamicRegistries.streamAllRegistries().forEach(registryEntry -> {
+					if (registryEntry.key().getValue().getNamespace().equals("minecraft")) {
+						registryEntry.value().streamTags().forEach(tagKey -> {
+							// Grab tags under 'forge' namespace
```

Comment:
Copy-pasted comment? 😅

Reply:
Yep. Was making the two PRs side-by-side. Fixing now